extension/controller/experiments/TranslationBar/api.js

```diff
@@ -128,15 +128,40 @@ const translationNotificationManagers = new Map();
                 }
 
                 const notificationBox = tab.browser.ownerGlobal.gBrowser.getNotificationBox(tab.browser);
-                let notif = notificationBox.appendNotification("fxtranslation-notification", {
+                let notif = null;
+
+                /*
+                 * we need to check if the notificationBox.appendNotification
+                 * requires 7 positional arguments and instantiate it differently
+                 * in order to keep it compatible with older Fx versions
+                 * see: https://github.com/mozilla/firefox-translations/issues/363#issuecomment-1151022189
+                 */
+                if (notificationBox.appendNotification.length === 7) {
+                  notif = notificationBox.appendNotification(
+                  "fxtranslation-notification",
+                  "",
+                  "",
+                  notificationBox.PRIORITY_INFO_HIGH,
+                  null,
+                  () => {
+                    // removed / dismissed / disconnected in any way.
+                    translationNotificationManagers.delete(tabId);
+                    // ^ may also happen when the tab is navigated.
+                  },
+                  TRANSLATION_NOTIFICATION_ELEMENT_ID,
+                  );
```

Comment:
The former 7 parameters have been replaced with `aType, aNotification, aButtons`. Of these, only `aType, aNotification` are passed in practice. The mapping between the old and new syntax is:
1. aLabel = aNotification.label = unspecified
2. aValue = aType = "fxtranslation-notification"
3. aImage = (removed)
4. aPriority = aNotification.priority = notificationBox.PRIORITY_INFO_HIGH
5. aButtons = aButtons = unspecified
6. aEventCallback = aNotification.eventCallback = `() => { ... }`
7. aNotificationIs = aNotification.notificationIs = TRANSLATION_NOTIFICATION_ELEMENT_ID

Given this, the equivalent replacement should be as follows

```suggestion
                  notif = notificationBox.appendNotification(
                    null,
                    "fxtranslation-notification",
                    null,
                    notificationBox.PRIORITY_INFO_HIGH,
                    null,
                    () => {
                      // removed / dismissed / disconnected in any way.
                      translationNotificationManagers.delete(tabId);
                      // ^ may also happen when the tab is navigated.
                    },
                    TRANSLATION_NOTIFICATION_ELEMENT_ID,
                  );
```

An alternative that may be more readable/maintainable is:

```js
const notificationId = "fxtranslation-notification";
const priority = notificationBox.PRIORITY_INFO_HIGH;
const eventCallback = () => { ... };
const notificationIs = TRANSLATION_NOTIFICATION_ELEMENT_ID;

if (notificationBox.appendNotification.length === 7) {
  // Firefox 93 and earlier
  notif = notificationBox.appendNotification(
    null,
    notificationId,
    null,
    priority,
    null,
    eventCallback,
    notificationIs
  );
} else {
  // Firefox 94 and later
  notif = notificationBox.appendNotification(notificationId, {
    priority,
    eventCallback,
    notificationIs,
  });
}
```